CLAUDE.md

```
@@ -650,3 +650,72 @@ export default defineSchema({
 });
 ```
+
+## Next.js + Convex Prefetch Guidelines
+
+### Why use `preloadQuery`?
+- By default, Next.js Client Components don’t block on Convex data and render in a "loading" state.
+- `preloadQuery` allows you to load data on the server (Server Components) and pass it to the client pre-hydrated.
+- This combines **fast server rendering** with **reactive real-time updates** once on the client.
+
+### Example: Server → Client Wrapper
+```ts
```

markdownlint-cli2 (0.18.1)

Code block style
Expected: indented; Actual: fenced

(MD046, code-block-style)